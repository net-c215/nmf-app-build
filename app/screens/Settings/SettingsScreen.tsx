import React from "react";
import { ScrollView } from "react-native";
import { Button, Text, SocialMedia } from "../../components";
import { SettingsRow } from "./components";
import * as WebBrowser from "expo-web-browser";
import styles from "./SettingsScreen.styles";
import navigationOptions from "./SettingsScreen.navigationOptions";
import { t } from "../../utils";

interface Props {
  navigation: {
    push: (screen: string) => void;
  };
}

const SettingsScreen = ({ navigation }: Props) => {
  const rowItems = [
    {
      title: t("SETTINGS_SCREEN_ABOUT"),
      onPress: () => navigation.push("About")
    },
    {
      title: t("SETTINGS_SCREEN_NMF_EARTH"),
      onPress: () => WebBrowser.openBrowserAsync("http://nmf.earth")
    },
    {
      title: t("SETTINGS_SCREEN_ROADMAP"),
      onPress: () =>
        WebBrowser.openBrowserAsync(
          "https://www.notion.so/notmyfault/cb98dacb015f4a18a7ebac5c0319495b?v=da9ec9ce3096417186dfd229a82aa90e"
        )
    },
    {
      title: t("SETTINGS_SCREEN_FEEDBACK"),
      onPress: () =>
        WebBrowser.openBrowserAsync("https://nmf-earth.typeform.com/to/w0nNSk")
    },
    {
      title: t("SETTINGS_SCREEN_TERMS_OF_USE"),
      onPress: () =>
        WebBrowser.openBrowserAsync("http://nmf.earth/terms-of-use.pdf")
    }
  ];

  return (
    <ScrollView style={styles.container}>
      {rowItems.map((item, index) => (
        <SettingsRow
          key={index}
          onPress={item.onPress}
          title={item.title}
          isLastItem={index === rowItems.length - 1}
        />
      ))}

      <SocialMedia />

      <Button.Primary
        style={styles.hiddenBtn}
        textType={"Primary"}
        onPress={() => navigation.push("Storybook")}
      >
        <Text.Primary white center>
          Open Storybook
        </Text.Primary>
      </Button.Primary>
    </ScrollView>
  );
};

SettingsScreen.navigationOptions = navigationOptions;

export default SettingsScreen;
