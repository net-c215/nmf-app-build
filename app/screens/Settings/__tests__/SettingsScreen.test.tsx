import React from "react";
import renderer from "react-test-renderer";
import SettingsScreen from "../SettingsScreen";

let props;

it("SettingsScreen renders correctly", () => {
  const tree = renderer.create(<SettingsScreen {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
