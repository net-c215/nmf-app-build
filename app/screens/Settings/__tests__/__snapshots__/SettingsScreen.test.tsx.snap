// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsScreen renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "paddingTop": 16,
    }
  }
>
  <View>
    <SettingsRow
      isLastItem={false}
      onPress={[Function]}
      title="About"
    />
    <SettingsRow
      isLastItem={false}
      onPress={[Function]}
      title="NMF.earth"
    />
    <SettingsRow
      isLastItem={false}
      onPress={[Function]}
      title="Roadmap"
    />
    <SettingsRow
      isLastItem={false}
      onPress={[Function]}
      title="Feedback"
    />
    <SettingsRow
      isLastItem={true}
      onPress={[Function]}
      title="Terms of use"
    />
    <SocialMedia />
    <Button.Primary
      onPress={[Function]}
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
      textType="Primary"
    >
      <Text.Primary
        center={true}
        white={true}
      >
        Open Storybook
      </Text.Primary>
    </Button.Primary>
  </View>
</RCTScrollView>
`;
